controllers/controllers/workloads/cforg_controller_test.go

```diff
@@ -0,0 +1,415 @@
+
+				It("should requeue", func() {
+					Expect(reconcileResult).To(Equal(ctrl.Result{RequeueAfter: 100 * time.Millisecond}))
+					Expect(reconcileErr).NotTo(HaveOccurred())
+
+					// validate call counts
+					Expect(fakeClient.GetCallCount()).To(Equal(2))
+					Expect(fakeClient.CreateCallCount()).To(Equal(0))
+					Expect(fakeClient.PatchCallCount()).To(Equal(0))
+					Expect(fakeStatusWriter.UpdateCallCount()).To(Equal(0))
+				})
+			})
+
+			When("fetch namespace returns an error ", func() {
+				BeforeEach(func() {
+					cfOrgError = nil
+					subNamespaceAnchorError = nil
+					subNamespaceAnchor.Status.State = v1alpha2.Ok
+					namespaceError = errors.New(failsOnPurposeErrorMessage)
+				})
+
+				It("should requeue", func() {
+					Expect(reconcileResult).To(Equal(ctrl.Result{RequeueAfter: 100 * time.Millisecond}))
+					Expect(reconcileErr).NotTo(HaveOccurred())
+
+					// validate call counts
+					Expect(fakeClient.GetCallCount()).To(Equal(3))
+					Expect(fakeClient.CreateCallCount()).To(Equal(0))
+					Expect(fakeClient.PatchCallCount()).To(Equal(0))
+					Expect(fakeStatusWriter.UpdateCallCount()).To(Equal(0))
+				})
+			})
+
+			When("set cascading delete returns an error ", func() {
+				BeforeEach(func() {
+					cfOrgError = nil
+					subNamespaceAnchorError = nil
+					subNamespaceAnchor.Status.State = v1alpha2.Ok
+					namespaceError = nil
+					fakeClient.PatchReturns(errors.New(failsOnPurposeErrorMessage))
+				})
+
+				It("should return an error", func() {
+					Expect(reconcileErr).To(MatchError(ContainSubstring(failsOnPurposeErrorMessage)))
+
+					// validate call counts
+					Expect(fakeClient.GetCallCount()).To(Equal(3))
+					Expect(fakeClient.CreateCallCount()).To(Equal(0))
+					Expect(fakeClient.PatchCallCount()).To(Equal(1))
+					Expect(fakeStatusWriter.UpdateCallCount()).To(Equal(0))
+				})
+			})
+
+			When("update CFOrg status to ready returns an error", func() {
+				BeforeEach(func() {
+					cfOrgError = nil
+					subNamespaceAnchorError = nil
+					subNamespaceAnchor.Status.State = v1alpha2.Ok
+					namespaceError = nil
+					fakeClient.PatchReturns(nil)
+					fakeStatusWriter.UpdateReturns(errors.New(failsOnPurposeErrorMessage))
+				})
+
+				It("should return an error", func() {
+					Expect(reconcileErr).To(MatchError(failsOnPurposeErrorMessage))
+
+					// validate call counts
+					Expect(fakeClient.GetCallCount()).To(Equal(3))
+					Expect(fakeClient.CreateCallCount()).To(Equal(0))
+					Expect(fakeClient.PatchCallCount()).To(Equal(1))
+					Expect(fakeStatusWriter.UpdateCallCount()).To(Equal(1))
+				})
+			})
+		})
+	})
+
+	When("a CFOrg is being deleted", func() {
+		BeforeEach(func() {
+			cfOrgError = nil
+			cfOrg.ObjectMeta.DeletionTimestamp = &metav1.Time{
+				Time: time.Now(),
+			}
+		})
+		It("returns an empty result and does not return error", func() {
+			Expect(reconcileResult).To(Equal(ctrl.Result{}))
+			Expect(reconcileErr).NotTo(HaveOccurred())
+
+			// validate call counts
```

Comment:
as there is no specific code that handles delete, I don't think that further verifications after the error check make much sense

Reply:
Done